src/dos/cdrom.h

```diff
@@ -354,18 +349,42 @@ class CDROM_Interface_Image final : public CDROM_Interface
 };
 
 #if defined (LINUX)
-class CDROM_Interface_Ioctl : public CDROM_Interface_SDL
-{
+class CDROM_Interface_Ioctl : public CDROM_Interface {
 public:
-	CDROM_Interface_Ioctl		();
-
-	bool	SetDevice		(const char* path, const int cd_number);
-	bool	GetUPC			(unsigned char& attr, char* upc);
-	bool	ReadSectors		(PhysPt buffer, const bool raw, const uint32_t sector, const uint16_t num);
-	bool	ReadSectorsHost	(void* buffer, bool raw, unsigned long sector, unsigned long num);
+	~CDROM_Interface_Ioctl();
+
+	bool SetDevice(const char* path, const int cd_number);
+	bool GetUPC(unsigned char& attr, char* upc);
+	bool GetAudioTracks(uint8_t& stTrack, uint8_t& end, TMSF& leadOut);
+	bool GetAudioTrackInfo(uint8_t track, TMSF& start, unsigned char& attr);
+	bool GetAudioSub(unsigned char& attr, unsigned char& track,
+	                 unsigned char& index, TMSF& relPos, TMSF& absPos);
+	bool GetAudioStatus(bool& playing, bool& pause);
+	bool GetMediaTrayStatus(bool& mediaPresent, bool& mediaChanged,
+	                        bool& trayOpen);
+	bool ReadSectors(PhysPt buffer, const bool raw, const uint32_t sector,
+	                 const uint16_t num);
+	bool ReadSectorsHost(void* buffer, bool raw, unsigned long sector,
+	                     unsigned long num);
+	bool PlayAudioSector(const uint32_t start, uint32_t len);
+	bool PauseAudio(bool resume);
+	bool StopAudio();
+	void ChannelControl(TCtrl ctrl);
+	bool LoadUnloadMedia(bool unload);
 
 private:
-	char	device_name[512];
+	void CDAudioCallback(uint16_t requested_frames);
+	void InitAudio();
+
+	int cdrom_fd                            = -1;
+	mixer_channel_t mixer_channel           = nullptr;
+	std::unique_ptr<int16_t[]> audio_buffer = nullptr;
+	size_t audio_position                   = 0;
+	size_t frames_in_buffer                 = 0;
+	int current_sector                      = 0;
+	int sectors_remaining                   = 0;
```

Comment:
`uint16_t` for both (or the smallest integer we can use).

Reply:
It definitely can't be `uint16_t` since we have this constant `#define MAX_REDBOOK_SECTOR 1826090u`.  Could change it to be unsigned since negative values don't make sense.  I just left it `int` because that's ultimately what the `ioctl` returns and takes for sector values.

Reply:
Yeah, we try to do better than the loose C way of doing things, but probably ok for this specific case.